x-pack/plugins/data_visualizer/public/application/common/components/stats_table/utils.ts

```diff
@@ -5,7 +5,6 @@
  * 2.0.
  */
 
-import { getBreakpoint } from '@elastic/eui';
```

Comment:
Is it possible to make use of `useCurrentEuiBreakpoint` here so we don't have to use hard coded widths like `(width <= 575)` further down?

Reply:
Unfortunately, no. I tried it in https://github.com/elastic/kibana/pull/140323/commits/4e60ce281de1ae58d14664bab172e04dba483d72, but this caused several functional tests to fail (correctly so). I had to revert the commit and do https://github.com/elastic/kibana/pull/140323/commits/2aa2234a9286617752db111cdf1836eea94e4561 instead.

The reason why is because `useCurrentEuiBreakpoint` only observes `window.innerWidth` and does not accept variable widths like `getBreakpoint`. Unfortunately, `DataVisualizerTable` is not always the width of the screen; it is occasionally in panels that are (e.g.) half the size of the screen or less.

So this logic does not translate to the new breakpoint hooks; but to be honest the concept of a breakpoint does tie in to windows/screens and not necessarily arbitrary component sizes/container queries, so the switch usage is perfectly valid - it just happens to use widths that correspond to EUI's breakpoint sizes (which you could easily change later).

Hope that makes sense, if not LMK and I can tag you into a Slack thread that had more discussion on this.

Reply:
I'm not sure I explained well. You **cannot** migrate your current resize observer / `calculateTableColumnsDimensions` logic to `useCurrentEuiBreakpoint` because your table width is not always 1:1 with `window.innerWidth`.

At most you can move away the static numbers in your switch statement to not be 1:1 with EUI's breakpoint sizes so it's clear you're **not** trying to use our breakpoints.

cc @qn895 on this as we discussed this while I was working on the deprecation, hoping you can help clarify!

Reply:
Quynh pinged me about it and I suggested this: https://github.com/elastic/kibana/pull/140323/commits/e6be71900d60d5718ffb910c301f5eef089e984d

Using a static map/constant and changing the breakpoints slightly from EUI may affect functionality slightly compared to before but helps more clearly distinguish that this table uses its own component-width-based breakpoints, and not EUI's screen-size-based breakpoints.